authhelper.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/muir/nfigure"
+	"github.com/pkg/browser"
+)
+
+type Output struct {
+	ExpiresAt          string `json:"expire"` // RFC3339 format
+	Password           string `json:"password"`
+	Username           string `json:"username"`
+	Email              string `json:"email,omitempty"`
+	ModelVersionNumber int    `json:"modelVersionNumber"`
+}
+
+const baseURL = "https://singlestore.com/engine-sso"
```

Comment:
does this exist, or is that what I need to implement?  If we want it tied to portal auth, we'd want the `portal` subdomain instead: `https://portal.singlestore.com/engine-sso`

Reply:
Yeah, that's what you need to create.  I'll change the URL.

Reply:
I pass the reply URL to the FE.
It's the `returnTo` parameter.  You must verify that it starts with "http://localhost:"